### PR TITLE
Misty/filters

### PR DESCRIPTION
**Pull request by mistyb01**
## What
The UI components for the filtering menu, as seen in the Projects page mockup.

This PR contains these components:
- `SearchToggle.tsx`: The search icon that you click to toggle a search bar.
- `FilterDropdown.tsx`: A select component. All of its values are given to it via props.

And a couple of containers:
- `SearchContainer.tsx`: A responsive container for the whole search/filtering section
- `FilterContainer.tsx`: Mostly for testing purposes-- contains state and event handlers that can be moved to the actual projects page.

## How Do
The design of the dropdowns are the MUI default, rather than the one shown in the figma, since I struggled with the customization. Kylie said this looks fine, but if anyone more comfortable with MUI wants to give it a shot, feel free!

### Test code
Add these imports to any page:
```
import FilterContainer from '../components/filters/FilterContainer'
import SearchContainer from '../components/filters/SearchContainer'
import SearchToggle from '../components/filters/SearchToggle'
```

then add this jsx to the body:
```
<SearchContainer>
     <SearchToggle/>
     <FilterContainer/>
</SearchContainer>
```

## Show Me
Desktop:
image

Mobile:
image

### CODE REVIEW

File: components/filters/FilterContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  const handleRoleChange = (event: SelectChangeEvent) => {
+    setRoleSelection(event.target.value as string)
+  }
+
+  const handleProgramChange = (event: SelectChangeEvent) => {
+    setProgramSelection(event.target.value as string)
+  }
+
+  return (
```
Comment (wheninseattle) [inline components/filters/FilterContainer.tsx:18]: The Figma mobile sketch is showing the two FilterDropdowns in a row, but it appears to not display correctly on a mobile phone (chrome dev tools). Do we want to shift it to a column at the xs breakpoint? 
    direction={{xs: 'column', md: 'row'}}

**Review by wheninseattle — APPROVED** [review]
Looks great! 😊

File: components/filters/SearchToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'))
+
+  const [displayInput, setDisplayInput] = useState(isMobile)
+  const [query, setQuery] = useState('')
+
+  // useRef to target the TextField input, so we can make it auto-focus.
+  const textInput = useRef(null)
+
+  // For setting the ref to the TextField when it appears in the DOM.
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (mistyb01) [reply components/filters/SearchToggle.tsx:22]: That's totally fine! The useEffect & the useRef code is a mix of different stackoverflow answers, after all.. 😆 
But this code is for a minor UX thing-- I wanted the search bar's input to automatically be focused on when the user toggles it! 

To do that, I needed the useRef hook to be able to change a DOM element's attributes, which in this case, was the input element's focus.
And the useEffect is for detecting whenever the "displayInput" state changes, so I can change the focus.

So yeah, kinda complicated code for a simple thing. There might be a better way!

File: components/filters/FilterContainer.tsx (hunk 1 above)
Comment (mistyb01) [reply components/filters/FilterContainer.tsx:18]: I just made that change. Thanks for catching that!

**Pull request merged**